lib/util/git/private-key.ts

```diff
@@ -87,7 +87,7 @@ class SSHKey extends PrivateKey {
     if (await this.hasPassphrase(keyFileName)) {
       throw new Error('SSH key must have an empty passhprase');
     }
-    await fs.outputFile(keyFileName, this.key);
+    await fs.outputFile(keyFileName, this.key + '\n');
```

Comment:
```suggestion
    await fs.outputFile(keyFileName, this.key.replace(/\n?$/, '\n'));
```

This change will mean that the newline is appended only if it's missing

Reply:
Both these changes do pass local testing of but running it on a actual repository fails. I've updated the PR body with my testcase so this can be reproduced.